plugins/woocommerce/includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -137,6 +174,18 @@ public function create( &$product ) {
 
 		if ( $id && ! is_wp_error( $id ) ) {
 			$product->set_id( $id );
+			$sku = $product->get_sku();
+
+			/**
+			 * If SKU is already under processing aka Duplicate SKU
+			 * because of concurrent requests, then we should not proceed
+			 * Delete the product and throw an exception.
+			 */
+			if ( ! empty( $sku ) && ! $this->obtain_lock_on_sku_for_concurrent_requests( $product ) ) {
+				$product->delete();
```

Comment:
```suggestion
				$product->delete( true );
```

I noticed after running the test script that the products that were deleted because they had the same SKU were sitting in the trash. Would it make sense to delete them permanently instead so they don't continue taking up space in the db?

Reply:
Good point, made this change